pkg/manager/member/tikv_scaler.go

```diff
@@ -59,31 +60,55 @@ func (s *tikvScaler) ScaleOut(meta metav1.Object, oldSet *apps.StatefulSet, newS
 		return nil
 	}
 
+	ls, err := label.New().Instance(tc.Name).Component(label.TiKVLabelVal).Selector()
+	if err != nil {
+		return fmt.Errorf("invalid label selector: %w", err)
+	}
+
+	pods, err := s.deps.PodLister.Pods(tc.Namespace).List(ls)
+	if err != nil {
+		return fmt.Errorf("can't list pods with label selector %s: %w", ls, err)
+	}
+
+	if len(pods) < int(*oldSet.Spec.Replicas) {
+		return fmt.Errorf("wait until tikv pods are created, expected in sts %v, current pods num %v", *oldSet.Spec.Replicas, len(pods))
+	}
+
+	unscheduledPods := sets.NewString()
+	for _, pod := range pods {
+		if pod.Spec.NodeName == "" {
+			unscheduledPods.Insert(pod.Name)
+		}
+	}
+
+	if unscheduledPods.Len() > 0 {
+		return fmt.Errorf("wait until tikv pods are scheduled, unscheduled pods: %v", unscheduledPods)
+	}
+
 	scaleOutParallelism := tc.Spec.TiKV.GetScaleOutParallelism()
 	_, ordinals, replicas, deleteSlots := scaleMulti(oldSet, newSet, scaleOutParallelism)
 	klog.Infof("scaling out tikv statefulset %s/%s, ordinal: %v (replicas: %d, scale out parallelism: %d, delete slots: %v)",
 		oldSet.Namespace, oldSet.Name, ordinals, replicas, scaleOutParallelism, deleteSlots.List())
 
 	var (
-		errs                         []error
-		finishedOrdinals             = sets.NewInt32()
-		updateReplicasAndDeleteSlots bool
+		errs             []error
+		finishedOrdinals = sets.NewInt32()
 	)
 	for _, ordinal := range ordinals {
 		err := s.scaleOutOne(tc, ordinal)
 		if err != nil {
 			errs = append(errs, err)
 		} else {
 			finishedOrdinals.Insert(ordinal)
-			updateReplicasAndDeleteSlots = true
 		}
 	}
-	if updateReplicasAndDeleteSlots {
-		setReplicasAndDeleteSlotsByFinished(scalingOutFlag, newSet, oldSet, ordinals, finishedOrdinals)
-	} else {
-		resetReplicas(newSet, oldSet)
```

Comment:
I'm not sure whether `resetReplicas(newSet, oldSet)` is needed for `return controller.RequeueErrorf("tikv.ScaleOut, cluster %s/%s ready to scale out, wait for next round", tc.GetNamespace(), tc.GetName())` in L122.